### ISSUE TITLE
LLVM+Clang: upgrade to version 15.0.4

### ISSUE THREAD

**Issue by agarny** [opencor/opencor#2675]

**Comment by agarny** [comment]
Giving up on upgrading to LLVM+Clang 15.0.4. For some reasons, on:
- Windows debug, we get an error message that reads `fatal error C1083: Cannot open compiler generated file: '': Invalid argument`; and on
- Linux, we get some undefined references (for `shm_open()` and `shm_unlink()` when linking the llibrary.

These issues may be indirectly linked to the fact that LLVM+Clang is built as one big shared library by OpenCOR and what used to work fine before may not work fine anymore...?

Either way, at the end of the day and for our purpose, to upgrade to LLVM+Clang 15.0.4 is not critical, so we may as well stick to the 14.x branch and go for the latest version on that on branch, i.e. 14.0.6 (see issue #2692).

**Comment by agarny** [comment]
FWIW, the Windows error message (`fatal error C1083: Cannot open compiler generated file: '': Invalid argument`) was due to Windows (!!) not liking long command line. The OpenCOR folder was `C:\Users\Alan\Desktop\OpenCOR-Debug`, but to move it to `C:\OCD` did the trick. Windows!

Still, even if we were able to solve the Linux issue (and it should be possible), we are still better off using 14.0.6. This is the last patch in a major version, so it ought to be safe and we (most probably) don't need the new features introduced in 15.x.
